src/pixelator/pixeldataset/datastores.py

```diff
@@ -457,37 +518,33 @@ def write_colocalization(self, colocalization: pd.DataFrame) -> None:
     def write_precomputed_layouts(
         self,
         layouts: Optional[PreComputedLayouts],
-        collapse_to_single_dataframe: bool = False,
     ) -> None:
         """Write pre-computed layouts to the data store."""
         if layouts is None:
             logger.debug("No layouts to write, will skip.")
             return
 
+        logger.debug("Starting to write layouts...")
+
+        self._set_to_write_mode()
         self._check_if_writeable(self.LAYOUTS_KEY)
 
-        logger.debug("Starting to write layouts...")
-        # This option is in place to allow collecting all the layouts into
-        # as single dataframe before writing (they will still be written into
-        # partitions), but this is much faster than writing them one by one
-        # for scenarios with many very small layouts.
-        if collapse_to_single_dataframe:
-            logger.debug("Writing from a single dataframe...")
-            self.write_dataframe(
-                layouts.to_df(),
-                self.LAYOUTS_KEY,
-                partitioning=PreComputedLayouts.DEFAULT_PARTITIONING,
-            )
-        else:
-            logger.debug("Writing by iterating components...")
-            for idx, layouts_to_write in enumerate(layouts.component_iterator()):
-                if idx % 100 == 0:
-                    logger.debug("Writing layouts...")
-                self.write_dataframe(
-                    layouts_to_write,
-                    self.LAYOUTS_KEY,
-                    partitioning=PreComputedLayouts.DEFAULT_PARTITIONING,
-                )
+        # This is a work around for the fact that sinking into parquet files
+        # from multiple sources is not supported. We therefore do this somewhat
+        # round about thing of first writing the parquet files to
+        # a temporary directory and then zipping them into the .pxl file.
+        with TemporaryDirectory(prefix="pixelator-") as tmp_dir:
```

Comment:
TemporaryDirectory() will write to the default TMP_DIR of the system, right? Do we need to do any cleanup() or the contet manager takes care of that?

Reply:
The context manager takes care of cleaning that up when it exits.